server/utils/tokenHandler.utils.ts

```diff
@@ -0,0 +1,30 @@
+import jwt from "jsonwebtoken";
+import envProvider from "./envProvider.utils";
+
+export interface IgenerageTokenInput {
+  userId: string;
+}
+
+interface ItokenHandler {
+  generageToken: (data: IgenerageTokenInput) => string;
+  verifyToken: (token: string) => jwt.JwtPayload | string;
+}
+
+const tokenHandler: ItokenHandler = {
+  generageToken: ({ userId }) => {
+    if (!envProvider.jwtSecret) throw Error("Invalid token secret");
+    const token = jwt.sign({ userId }, envProvider.jwtSecret, {
+      expiresIn: "24h",
+    });
+    return token;
+  },
+  verifyToken: async (token) => {
+    if (!envProvider.jwtSecret) throw Error("Invalid token secret");
+    try {
+      return await jwt.verify(token, envProvider.jwtSecret);
```

Comment:
why do we use await here? there is no need for that

Reply:
updated